library/std/src/sys/unix/os.rs

```diff
@@ -586,6 +586,36 @@ pub fn temp_dir() -> PathBuf {
     crate::env::var_os("TMPDIR").map(PathBuf::from).unwrap_or_else(|| {
         if cfg!(target_os = "android") {
             PathBuf::from("/data/local/tmp")
+        } else if cfg!(target_vendor = "apple") {
+            extern "C" {
+                fn confstr(
+                    name: libc::c_int,
+                    buf: *mut libc::c_char,
+                    len: libc::size_t,
+                ) -> libc::size_t;
+            }
+            let tmpdir = unsafe { libc::getenv(b"TMPDIR".as_ptr() as *const libc::c_char) };
```

Comment:
We already checked the env above. Also, don't use `libc::getenv` as it doesn't hold the env lock. Also, this string isn't NUL-terminated.